packages/plugin/src/code-gen/message-type-generator.ts

```diff
@@ -1,167 +1,175 @@
 
 
-    private readonly wellKnown: WellKnownTypes;
-    private readonly googleTypes: GoogleTypes;
-    private readonly typeMethodCreate: Create;
-    private readonly typeMethodInternalBinaryRead: InternalBinaryRead;
-    private readonly typeMethodInternalBinaryWrite: InternalBinaryWrite;
-    private readonly fieldInfoGenerator: FieldInfoGenerator;
-
-
-    constructor(symbols: SymbolTable, registry: DescriptorRegistry, imports: TypeScriptImports, comments: CommentGenerator, interpreter: Interpreter,
-                private readonly options: {
-                    runtimeImportPath: string;
-                    normalLongType: LongType;
-                    oneofKindDiscriminator: string;
-                    useProtoFieldName: boolean;
-                }) {
-        super(symbols, registry, imports, comments, interpreter);
-        this.fieldInfoGenerator = new FieldInfoGenerator(this.registry, this.imports, this.options);
-        this.wellKnown = new WellKnownTypes(this.registry, this.imports, this.options);
-        this.googleTypes = new GoogleTypes(this.registry, this.imports, this.options);
-        this.typeMethodCreate = new Create(this.registry, this.imports, this.interpreter, this.options);
-        this.typeMethodInternalBinaryRead = new InternalBinaryRead(this.registry, this.imports, this.interpreter, this.options);
-        this.typeMethodInternalBinaryWrite = new InternalBinaryWrite(this.registry, this.imports, this.interpreter, this.options);
+  private readonly wellKnown: WellKnownTypes;
+  private readonly googleTypes: GoogleTypes;
+  private readonly typeMethodCreate: Create;
+  private readonly typeMethodInternalBinaryRead: InternalBinaryRead;
+  private readonly typeMethodInternalBinaryWrite: InternalBinaryWrite;
+  private readonly fieldInfoGenerator: FieldInfoGenerator;
+  private readonly symbolKindClass = 'message-class';
+
+
+  constructor (symbols: SymbolTable, registry: DescriptorRegistry, imports: TypeScriptImports, comments: CommentGenerator, interpreter: Interpreter,
+    private readonly options: {
+      runtimeImportPath: string;
+      normalLongType: LongType;
+      oneofKindDiscriminator: string;
+      useProtoFieldName: boolean;
+    }) {
+    super(symbols, registry, imports, comments, interpreter);
+    this.fieldInfoGenerator = new FieldInfoGenerator(this.registry, this.imports, this.options);
+    this.wellKnown = new WellKnownTypes(this.registry, this.imports, this.options);
+    this.googleTypes = new GoogleTypes(this.registry, this.imports, this.options);
+    this.typeMethodCreate = new Create(this.registry, this.imports, this.interpreter, this.options);
+    this.typeMethodInternalBinaryRead = new InternalBinaryRead(this.registry, this.imports, this.interpreter, this.options);
+    this.typeMethodInternalBinaryWrite = new InternalBinaryWrite(this.registry, this.imports, this.interpreter, this.options);
+  }
+
+  registerSymbols (source: TypescriptFile, descriptor: DescriptorProto): void {
```

Comment:
Delta is confusing because of file formatting. Added registerSymbols function here and calling it below.